testament/testament.nim

```diff
@@ -506,7 +699,13 @@ proc logToBackend(param: ReportParams) =
 
 proc addResult(r: var TResults, param: ReportParams, givenSpec: ptr TSpec) =
   ## Report results to backend, end user (write to command-line) and so on.
-  
+
+  # REFACTOR: first of all, name: it does not /add/ it fucking *writes*
+  # output. "write" and "add" are two different things. Second: determine
+  # why `givenSpec` is an optional in the first place -- aren't we printing
+  # out results of the test? And each test has a spec attached to it.
```

Comment:
I remember this now... the issue is that `addResult` is more like add whatever stuff happened, and occassionally, that doesn't involve a test/spec... I can't quite find the call site(s) at the moment.